### PR TITLE
STYLE/CI: implement incremental linting for asv benchmarks

### PR DESCRIPTION
**Pull request by jbrockmendel**
Another try at #18620 

@mroeschke since it looks like you are going through the asv files one-by-one, the idea here is that when file X gets cleaned up, it can be un-excluded in the lint.sh command.

- [ ] closes #xxxx
- [ ] tests added / passed
- [ ] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [ ] whatsnew entry

### CODE REVIEW

**Comment by mroeschke** [comment]
👍 Sounds good to me @jbrockmendel

File: asv_bench/vbench_to_asv.py [hunk 1]
```diff
@@ -15,7 +15,8 @@ def vbench_to_asv_source(bench, kinds=None):
     output += tab + 'goal_time = 0.2\n\n'
 
     if bench.setup:
-        indented_setup = [tab * 2 + '{}\n'.format(x) for x in bench.setup.splitlines()]
+        indented_setup = [tab * 2 + '{}\n'.format(x)
```
Comment (gfyoung) [inline asv_bench/vbench_to_asv.py]: Slight preference for keyword arguments for clarity vs. bare `{}`.

**Comment by jorisvandenbossche** [comment]
@jbrockmendel No need to clean up the vbench_to_asv.py file (it is not used anymore, it is rather there for historical reasons. We could remove it, but let's do that in another PR). So I would revert the changes of that file here.

**Comment by jorisvandenbossche** [comment]
Thanks!

**Pull request merged**